### DISCUSSION TITLE
execute code in R console upon pressing enter

### DISCUSSION THREAD

**Discussion by benzipperer** [posit-dev/positron#5180] [Q&A] [answered]
Thanks for the awesome editor!

If I type code interactively in the R console, pressing enter sometimes starts a new line. I never want this to happen. Instead always want the code to run when I press enter. Is there a setting to change the default behavior? 

(I prefer the Rstudio default where enter executes and shift-enter starts a new line)

**Comment by lionel-** [comment]
Hi! The Positron console should work just like the RStudio one in that regard. Could you provide some code example or a video/screencast of this happening?

**Comment by benzipperer** [comment]
Sorry for the delay. Here's a screencast of me typing and then pressing Enter (not Shift + Enter):

![output](https://github.com/user-attachments/assets/98e10d97-2119-490d-8fad-79300ff4dcc7)

Not shown but if I force the cursor to the end of the line, by pressing End or using the right arrow, then Enter executes the code as intended. But note that in the screenshot above, where I don't do this, even if the cursor is at the end of the line, Enter will start a new line. 

If mid-line I press tab to accept a suggestion, then Enter executes the code as intended.

This is positron 2024.11.0 build 49 on Ubuntu 24.04. I confirmed the behavior persists if I launch positron with --disable-extensions.

**Reply by juliasilge** [reply]
Thank you for sharing this @benzipperer! We are not able to reproduce what you are seeing (<kbd>Enter</kbd> not sending the code to be executed), including on Linux, but certainly it is not the expected behavior. 😬

Can you tell us anything else about how you may be running Positron and/or R? Or how you installed Positron and/or R?

**Reply by benzipperer** [reply, marked as answer]
Thanks for trying! 

Could my problem be related to the setting "Editor: Accept Suggestion on Enter"? I had changed this to "off". Can you reproduce after changing that setting?

When that setting is off, I have the problem in the screencast above: typing `mtcars` then pressing ENTER starts a new line. Though it's not in the screencast, hitting ENTER a second time executes the code.

When that setting is on, I have a different problem: if I type `mtcars`, then ENTER "accepts" the suggestion of `mtcars` (which I had already typed anyways). Hitting ENTER a second time executes the code without starting a new line.

I'd prefer to just use TAB to accept suggestions and reserve ENTER to execute the code regardless of the suggestion status.

I installed positron on Ubuntu 24.04 using the deb package available on your github releases page via `sudo dpkg -i package.deb`. I installed R using [rig](https://github.com/r-lib/rig): `rig add next`, probably. Before replying to you I tried starting with a fresh installation of Positron: first I uninstalled Positron and deleted ~/.positron and ~/.config/Positron. Then I installed the latest Positron deb package 2024.11.0-116.

**Reply by lionel-** [reply]
Good hunch! I can reproduce what you see when that setting is off. I've filed an issue at https://github.com/posit-dev/positron/issues/5189

**Reply by benzipperer** [reply]
Maybe this is a separate issue, but one thing that might help users here is for positron to remove suggestions when they are already completed by the user when typing. 

For example, let's say you want to type in the console `mtcars` and execute it. When you begin to type `mtcars`, positron suggests `mtcars`. This is fine. However, if you complete typing mtcars, then positron doesn't "accept" your suggestion. At this point, if you want to run your code you have to hit ENTER twice: once to accept the suggestion, and then once more to execute the code.

In contrast Rstudio removes the `mtcars` suggestion once you finish typing mtcars.

**Reply by juliasilge** [reply]
I believe we have that tracked in https://github.com/posit-dev/positron/issues/550, if you would like to take a look.
